### ISSUE TITLE
Eliminate non-array opportunity keys and cutoffs in regional request

### ISSUE THREAD

**Issue by ansoncfit** [conveyal/r5#616]
From @abyrd: Regional request objects sent from the UI to the backend contain some obsolete fields now that we're sending arrays of cutoffs and destination pointsets. We can't simply remove the non-array fields, because presence or absence of these fields is currently used to detect the file storage version, which affects the S3 object naming.

Unless we want to indefinitely maintain code for handling several different file name and storage approaches, and record metadata about which format was used for each particular analysis, we will eventually want to 1) migrate the regional result database records and 2) normalize all S3 regional results to the most recent filename format (which includes the percentile and destination pointset ID). Once this is done we are free to remove obsolete fields in the UI-to-backend regional request class.

We have decided to that database migration will be deferred, but it will eventually be done. For this change as well as similar future changes, backward compatibility will still be maintained with older workers via AnalysisWorkerTasks, but UI-to-backend requests should always adhere to the most recent version.

By normalizing to the most recent results file name format, we mean adding destination point set ID and percentile to every results file name. See: https://github.com/conveyal/r5/blob/06bc4d3e61ed002671ee219a828779c90c308ff8/src/main/java/com/conveyal/analysis/controllers/RegionalAnalysisController.java#L287-L299

The effort required does not seem large, but the potential for damage is big as we rename every regional result on S3. Nonetheless we should not fear migration; avoiding migration doesn't solve anything and creates technical debt. So we want to get procedures in place to ensure we're not apprehensive about Mongo and S3 file migrations.

We could formally version the storage system but that's even more complex and costly than creating clean migrations.
